lottie/src/main/java/com/airbnb/lottie/LottieCompositionFactory.java

```diff
@@ -128,14 +129,19 @@ public static void setMaxCacheSize(int size) {
    * Parse an animation from raw/res. This is recommended over putting your animation in assets because
    * it uses a hard reference to R.
    * The resource id will be used as a cache key so future usages won't parse the json again.
+   * Note: to correctly load dark mode (-night) resources, make sure you pass Activity as a context (instead of e.g. the application context).
+   * The Activity won't be leaked.
    */
   public static LottieTask<LottieComposition> fromRawRes(Context context, @RawRes final int rawRes) {
     // Prevent accidentally leaking an Activity.
+    final WeakReference<Context> contextRef = new WeakReference<>(context);
     final Context appContext = context.getApplicationContext();
     return cache(rawResCacheKey(context, rawRes), new Callable<LottieResult<LottieComposition>>() {
       @Override
       public LottieResult<LottieComposition> call() {
-        return fromRawResSync(appContext, rawRes);
+        @Nullable Context originalContext = contextRef.get();
+        Context context = originalContext != null ? originalContext : appContext;
+        return fromRawResSync(context, rawRes);
```

Comment:
If the activity is in night mode and then gets dereferenced and you fall back to the application context, it might report day mode and store the night asset there.

Reply:
From what I observed, if the application context reports day mode (incorrectly), then the resources object from this context also returns the day asset (incorrectly). So in the case of Activity object not reachable, the flow is as follows:
1. The first call of `rawResCacheKey()` (the argument of the `cache` method) will return correct "night" key
1. In the `cache()` method, the callable is called, which obtains day resource from the application context and stores it under "day" cache key (re-generated from the application context in `fromRawResSync()`
1. Back in the `cache()` method, the callable result is stored in the cache under the "night" key - that's the only issue I can see here, I guess that the solution would be to change the callable signature from `Callable<LottieResult<LottieComposition>>` to something like `Callable<LottieResult<Pair<String /* new cache key* ,LottieComposition>>>` and store the result under new cache key passed in the `LottieResult`, but this would require changing the signature of `fromRawResSync()` method.

Reply:
@plnice Actually, I think we can remove the call where the `cache` function adds the composition to the cache. It is already done by `fromZipStreamSyncInternal` and `fromJsonReaderSyncInternal` so it is redundant there. If we do that then the `cache` function is just storing the task cache and to repro this bug, you would somehow have to join an in-progress task which seems highly improbable.

So the steps here would be:
1) Remove the the part of cache where it adds a composition to the cache
2) Wrap the existing cache.put calls in the 2 internal factory functions with `if (cacheKey != null)` (as it always should have been)

then I think this would be reasonable to land as-is.

Reply:
Got it! Added two commits with the changes.